starter_code/src/movies.js

```diff
@@ -1,20 +1,131 @@
 /* eslint no-restricted-globals: 'off' */
 // Turn duration of the movies from hours to minutes 
+function minutesCalc(str) {
+    var newArr = str.split(" ");
+    var hours = Number((newArr[0].split("h"))[0]);
+    if (newArr.length > 1) {
+        var duration = hours * 60 + Number((newArr[1].split("min"))[0]);
+    } else {
+        var duration = hours * 60;
+    }
+    return duration;
+}
 
+function turnHoursToMinutes(arr) {
+    var converMovies = arr.map(function (a) {
+        var b={};
+        for(key in a){
+            b[key] = a[key]                      
+        }
+        b.duration = minutesCalc(a.duration);          
+        return b;
+    })
+    return converMovies;
+}
 
 // Get the average of all rates with 2 decimals 
-
+function ratesAverage(arr) {
+    var total = arr.reduce(function (accumulator, item) {
+        return accumulator + Number(item.rate);
+    }, 0);
+    return parseFloat((total / arr.length).toFixed(2));
+};
 
 // Get the average of Drama Movies
-
+function dramaMoviesRate(arr) {
+    var dramaMovies = arr.filter(function (item) {
+        return item.genre.find(function (a) {return a === "Drama";}) 
+        === "Drama";
+    });
+    if (dramaMovies.length > 0) {
+        return ratesAverage(dramaMovies);
+    }
+}
 
 // Order by time duration, in growing order
-
+function orderByDuration(arr) {
+    //var newArr = turnHoursToMinutes(arr); <--- What?
+    return arr.sort(function(a,b) {
```

Comment:
This code passes the test (congratulations) but doesn't work exactly as expected. You are ordering durations that are strings, then you are ordering them alphabetically. This will be some piece of your sorted array:
```
...
{ title: 'Munna Bhai M.B.B.S.',
    year: '2003',
    director: 'Rajkumar Hirani',
    duration: '2h 36min',
    genre: [ 'Comedy', 'Drama', 'Romance' ],
    rate: '8.2' },
  { title: 'Stalker',
    year: '1979',
    director: 'Andrei Tarkovsky',
    duration: '2h 42min',
    genre: [ 'Drama', 'Sci-Fi' ],
    rate: '8.1' },
  { title: 'Jaws',
    year: '1975',
    director: 'Steven Spielberg',
    duration: '2h 4min',
    genre: [ 'Adventure', 'Drama', 'Thriller' ],
    rate: '8.0' },
  { title: 'The Best Years of Our Lives',
    year: '1946',
    director: 'William Wyler',
    duration: '2h 50min',
    genre: [ 'Drama', 'Romance', 'War' ],
    rate: '8.1' },
  { title: 'Dog Day Afternoon',
    year: '1975',
    director: 'Sidney Lumet',
    duration: '2h 5min',
    genre: [ 'Biography', 'Crime', 'Drama', 'Thriller' ],
    rate: '8.0' },
  { title: 'Jurassic Park',
    year: '2000',
    director: 'Steven Spielberg',
    duration: '2h 7min',
    genre: [ 'Adventure', 'Sci-Fi', 'Thriller' ],
    rate: '8.1' },
...
```
How do you solve this problem? Using before the method that you created for converting the durations into minutes.